src/test/java/seedu/workbook/model/internship/RoleTest.java

```diff
@@ -0,0 +1,41 @@
+package seedu.workbook.model.internship;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.workbook.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class RoleTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Role(null));
+    }
+
+    @Test
+    public void constructor_invalidName_throwsIllegalArgumentException() {
+        String invalidName = "";
+        assertThrows(IllegalArgumentException.class, () -> new Role(invalidName));
+    }
+
+    @Test
+    public void isValidName() {
+        // null name
+        assertThrows(NullPointerException.class, () -> Role.isValidRole(null));
+
+        // invalid name
+        assertFalse(Role.isValidRole("")); // empty string
+        assertFalse(Role.isValidRole(" ")); // spaces only
+        assertFalse(Role.isValidRole("^")); // only non-alphanumeric characters
+        assertFalse(Role.isValidRole("meta*")); // contains non-alphanumeric characters
```

Comment:
Might be nitpicking here but maybe change this to `frontend*` instead because meta is a company. I forgot what principle this is but in essence I think the String names in the testclasses should closely follow the context of its main class, similar to the comment I made below